licence-checker/licence-checker.py

```diff
@@ -43,17 +44,24 @@ def first_word(self):
 class CommentStyle:
     '''Base class for comment style objects'''
     def __init__(self, first_line_prefix, comment_prefix):
-        self.first_line_prefix = first_line_prefix
-        self.comment_prefix = comment_prefix
+        self.first_line_prefix = re.compile(re.escape(first_line_prefix))
+        self.comment_prefix = re.escape(comment_prefix)
+        self.comment_prefix_pattern = re.compile(self.comment_prefix)
```

Comment:
I think it would be cleaner to leave these as strings: we should generate regexes (or not) in `LicenceMatcher`, where we know what mode we're in.

Reply:
I see. I am asking myself if it wouldn't be better to sub-class the `LicenceMatcher` (or create a common parent class) and crate a `RegexLicenceMatcher`. If we fix the matching logic in the `LicenceMatcher` we will end up with a lot of code duplication such as:
```python
    def __init__(self, comment_style, licence, match_regex):
        self.style = comment_style
        if match_regex:
                ...
        else:
                ...
```
The same for the actual matching logic. We need a block for the non-regex and one for the regex.
Maybe I also misunderstood your point here, I apologize for being slow.

Reply:
I'm not sure whether the subclassing is needed. Yes, we'll have some if/else in the `LicenceMatcher` constructor, but I don't really see a problem with that.